pandas/core/dtypes/cast.py

```diff
@@ -1094,14 +1093,12 @@ def astype_nansafe(
         The dtype was a datetime64/timedelta64 dtype, but it had no unit.
     """
     if arr.ndim > 1:
-        # Make sure we are doing non-copy ravel and reshape.
-        flags = arr.flags
-        flat = arr.ravel("K")
```

Comment:
wont this mean a perf hit in cases where arr is F-contiguous?

Reply:
Yep (some discussion about that in the pr body). Would be a regression from master (but not from 1.2.x). The other fix for this regression is to just use `arr.ravel(order)` where `order` is the same as used for `reshape`.

That avoids the hit you're talking about, but doesn't fix the perf regression (although that perf regression is a bit suspicious, was trying to profile it, but on larger data sizes the regression doesn't show up anymore)

Reply:
does `arr.ravel(order)`  fix this?

Reply:
Yeah `arr.ravel(order)` fixes the regression in behavior, but not the performance regression (discussed a bit in the comments above). Seems like it comes down to computations like `bottleneck.reduce.nanmean` being slower on F-contiguity data.

Reply:
> Seems like it comes down to computations like bottleneck.reduce.nanmean being slower on F-contiguity data.

yes for sure that would be true.

ok so the question remains, this goes back to 1.2.x perf? are the f-contig routines hit? when do we hit these

Reply:
Yep exactly -> the benchmarks shown at the top of the pr are back to 1.2.x perf. With the perf regression on master, for those benchmarks we hit reduction routines with f-contig data. With this pr and 1.2.x, we hit those routines with C-contig data.